### DISCUSSION TITLE
App Router always load server export from library

### DISCUSSION THREAD

**Discussion by FlorianKoerner** [vercel/next.js#56864] [Help]
### Summary

I am the author of the [DiceBear](https://github.com/dicebear/dicebear) library and I am increasingly being contacted by next.js users who are having problems in connection with the new App Router.

The problem is that next.js, in conjunction with the App Router, only loads the "server" export from the [@dicebear/converter](https://github.com/dicebear/dicebear/tree/7.x/packages/%40dicebear/converter) package and completely ignores that it may be a browser environment.

The problem also exists when "client only" is specified. Only the server export is still used. I don't know what the problem is - do you have any ideas?

- https://github.com/hustLer2k/dicebear-browser/issues/1#issuecomment-1453758866
- https://github.com/dicebear/dicebear/issues/357
- https://github.com/orgs/dicebear/discussions/324
- https://github.com/dicebear/dicebear/issues/383

### Additional information

```js
Error message:


> my-app@0.1.0 build
> next build

Failed to compile.

./node_modules/@dicebear/converter/lib/node/core.js
Module not found: Can't resolve '@resvg/resvg-js'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@dicebear/converter/lib/node/index.js
./node_modules/@dicebear/core/lib/core.js
./node_modules/@dicebear/core/lib/index.js
./src/app/page.tsx

./node_modules/@dicebear/converter/lib/node/core.js
Module not found: Can't resolve 'sharp'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@dicebear/converter/lib/node/index.js
./node_modules/@dicebear/core/lib/core.js
./node_modules/@dicebear/core/lib/index.js
./src/app/page.tsx

./node_modules/@dicebear/converter/lib/node/core.js
Module not found: Can't resolve 'exiftool-vendored'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@dicebear/converter/lib/node/index.js
./node_modules/@dicebear/core/lib/core.js
./node_modules/@dicebear/core/lib/index.js
./src/app/page.tsx


> Build failed because of webpack errors
```
```


### Example

https://github.com/FlorianKoerner/dicebear-nextjs

**Comment by dclipca** [comment]
With `next.config.mjs` and `"next": "14.1.4"` (you have to `npm i @dicebear/converter`):

```ts
import path from 'path';
import { fileURLToPath } from 'url'; 

/** @type {import('next').NextConfig} */

// Convert the module URL to a directory path
const __dirname = path.dirname(fileURLToPath(import.meta.url));

const nextConfig = {
  webpack: (config, { isServer }) => {
    config.resolve.alias["@dicebear/converter"] = path.resolve(
      __dirname,
      "node_modules/@dicebear/converter/lib/index.js"
    );

    return config;
  },
};
```

**Reply by XoL1507** [reply]
i'm sorry.. i just try vercel to fully run. and made it.. but

**Reply by dclipca** [reply]
Wdym?
